### PR TITLE
Go over unit tests for all components

### LINKED ISSUE #197: Go over unit tests for all components and make sure they follow what the README says

### PR DESCRIPTION
**Pull request by raggeju**
closes #197

### CODE REVIEW

File: src/atoms/checkbox/Checkbox.test.tsx [hunk 1]
```diff
@@ -3,9 +3,24 @@ import userEvent from '@testing-library/user-event';
...
+			inputElement = getByRole(checkboxRole);
+		userEvent.click(inputElement);
+		expect(inputElement).toBeChecked();
+	});
+
+	test('It cannot be checked when disabled', () => {
+		const { getByRole } = render(<Checkbox id={checkboxId} disabled={true} />),
+			inputElement = getByRole(checkboxRole) as HTMLInputElement;
+		userEvent.click(inputElement);
+		expect(inputElement.checked).toEqual(false);
```
Comment (tobiasljungstrom) [inline src/atoms/checkbox/Checkbox.test.tsx:21]: I think you can write `not.toBeChecked()` or someting.

File: src/atoms/heading/Heading.test.tsx [hunk 2]
```diff
@@ -6,22 +6,20 @@ import Heading, { HeadingLevel } from './Heading';
...
+	test('It renders the correct html element', () => {
 		headings.forEach((level, index) => {
 			const { getByRole } = render(<Heading as={level}>I am heading</Heading>);
 			expect(getByRole('heading', { level: index + 1 })).toBeInTheDocument();
 		});
 	});
 
-	test('The heading contains children', () => {
+	test('It contains children', () => {
 		const { findByText } = render(<Heading as={'h1'}>I am heading</Heading>);
```
Comment (tomahl) [inline src/atoms/heading/Heading.test.tsx:17]:
```suggestion
		const { findByText } = render(<Heading as="h1">I am heading</Heading>);
```

File: src/atoms/heading/Heading.test.tsx [hunk 3]
```diff
@@ -6,22 +6,20 @@ import Heading, { HeadingLevel } from './Heading';
...
-
 		expect(findByText('I am heading')).toBeTruthy;
 	});
 
 	test('It passes automatic accesibility tests', async () => {
 		const { container } = render(<Heading as={'h1'}>I am heading</Heading>);
```
Comment (tomahl) [inline src/atoms/heading/Heading.test.tsx:22]:
```suggestion
		const { container } = render(<Heading as="h1">I am heading</Heading>);
```

File: src/molecules/search-field/SearchField.test.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+		expect(inputElement).toHaveFocus();
+	});
+
+	test('It gets a reset button when entering text in input element', () => {
+		const { container } = render(template),
+			inputElement = container.querySelector('input') as HTMLInputElement;
+
+		userEvent.type(inputElement, ' test2');
+
+		expect(container.querySelector('button')).toBeTruthy();
```
Comment (tobiasljungstrom) [inline src/molecules/search-field/SearchField.test.tsx:40]: https://github.com/testing-library/jest-dom#tobeinthedocument Might do the trick here.

**Review by tomahl — DISMISSED** [review]

**Review by tomahl — APPROVED** [review]

**Review by tobiasljungstrom — APPROVED** [review]

**Pull request merged**